ddprof-lib/src/main/cpp/profiler.cpp

```diff
@@ -414,6 +414,28 @@ int Profiler::getJavaTraceAsync(void* ucontext, ASGCT_CallFrame* frames, int max
      */
     bool in_java = (state == 8 || state == 9);
     if (in_java && java_ctx->sp != 0) {
+        // check the frame is alive if we have access to vmstructs
+        int hsversion = VM::hotspot_version();
+        if (hsversion >= 0 && hsversion < 21 && VMStructs::hasMethodStructs()) {
+            NMethod* nmethod = CodeHeap::findNMethod(java_ctx->pc);
+            bool produceErrorFrame = true;
+            if (nmethod != nullptr && nmethod->isNMethod()) {
+                if (nmethod->isAlive()) {
```

Comment:
so either we are alive, or we are killing zombie frames. What an interesting fun code :smile: 
Minor: we could encapsulate the logic as `is_alive_function` or `top_function_is_alive` ?